lib/datadog/tracing/contrib/action_pack/configuration/settings.rb

```diff
@@ -24,7 +26,18 @@ class Settings < Contrib::Configuration::Settings
               o.lazy
             end
 
-            option :exception_controller
+            # DEV-2.0: Breaking changes for removal.
+            option :exception_controller do |o|
+              o.on_set do |value|
+                if value
+                  Datadog::Core.log_deprecation do
+                    "Option `#{o.instance_variable_get(:@name)}` has been deprecated and will be removed, "\
+                    'the value provided is no longer effective.'
```

Comment:
It's good to mention why it's being deprecated ("what will happen if I remove `:exception_controller`?"):
```suggestion
                    "Option `#{o.instance_variable_get(:@name)}` is no longer required to will be removed."
```